gcp/live/dev/spec/network_spec.rb

```diff
@@ -34,38 +34,79 @@ def get_gpii_pods_labeled(key, value)
     .select{ |item| item.has_label(key, value) }
 end
 
-def kubectl(cmd)
+def kubectl(cmd, debug=false)
+  if debug
+    puts "kubectl #{cmd} 2>/dev/null"
+  end
+
   `kubectl #{cmd} 2>/dev/null`
 end
 
+def file_exists_on_pod?(pod, file)
+  `kubectl exec -n gpii -it #{pod.name} -- test -f #{file} 2>/dev/null`
+
+  $? == 0
+end
+
+def busybox_is_setup_on?(pod)
+  return file_exists_on_pod?(pod, '/tmp/busybox')
+end
+
+def setup_busybox_on(pod)
+  busybox_file = download_busybox()
+
+  kubectl("cp -n gpii #{busybox_file} #{pod.name}:/tmp/busybox")
+
+  kubectl("exec -n gpii -it #{pod.name} -- chmod +x /tmp/busybox")
+end
+
+def download_busybox(url='https://busybox.net/downloads/binaries/1.21.1/busybox-x86_64')
+  filename = "/tmp/#{File.basename(url)}"
+
+  unless File.exist?(filename)
+    system("curl #{url} > #{file}")
+
+    fail "Error: Unable to download busybox" if $? != 0
+  end
+
+  return filename
+end
+
 describe "Security:" do
   before :all do
     @couchdb_pods = get_gpii_pods_labeled("app", "couchdb")
     @preferences_pods = get_gpii_pods_labeled("app", "preferences")
     @flowmanager_pods = get_gpii_pods_labeled("app", "flowmanager")
     @all_pods = @couchdb_pods | @preferences_pods | @flowmanager_pods
 
-    @couchdb_pods.each do |pod|
-      kubectl("exec -n gpii -it #{pod.name} -- sh -c \"[ ! -f /bin/nc ] && apt-get update -y && apt-get install netcat -y\"")
+    @all_pods.each do |pod|
+      unless busybox_is_setup_on?(pod)
+        setup_busybox_on(pod)
+      end
     end
   end
 
-  context "Preferences Pod" do
+  after :all do
+    @all_pods.each do |pod|
+      kubectl("exec -n gpii -it #{pod.name} -- rm -f /tmp/busybox")
+    end
+  end
+
+  context "Preferences" do
     context "Container Port 8081" do
-      it "listens on port 8081" do
+      it "listens on that port" do
         @preferences_pods.each do |pod|
-          kubectl("exec -n gpii -it #{pod.name} -c preferences -- nc -z #{pod.ip} 8081")
+          kubectl("exec -n gpii -it #{pod.name} -c preferences -- sh -c '/tmp/busybox nc -w 1 #{pod.ip} 8081 </dev/null'")
```

Comment:
From Tyler:
>Adding `-w 1` seems good, but why is `-z` no longer needed? If it was replaced by `</dev/null`, is that clearer than `-z`? Was there another motivation for this change?

Reply:
`-z` didn't seem to be available on the busybox binary that we're downloading:

```
BusyBox v1.21.1 (2013-07-08 11:34:59 CDT) multi-call binary.

Usage: nc [-iN] [-wN] [-l] [-p PORT] [-f FILE|IPADDR PORT] [-e PROG]

Open a pipe to IP:PORT or FILE

        -e PROG Run PROG after connect
        -l      Listen mode, for inbound connects
                (use -l twice with -e for persistent server)
        -p PORT Local port
        -w SEC  Timeout for connect
        -i SEC  Delay interval for lines sent
        -f FILE Use file (ala /dev/ttyS0) instead of network
```

Reply:
@mrtyler - it looks like `-z` is available on a later version of busybox. I'll update the url.